src/diff.jl

```diff
@@ -370,9 +370,15 @@ function expand_derivatives(O::Symbolic, simplify=false; throw_no_derivative=fal
         return simplify ? SymbolicUtils.simplify(O1) : O1
     end
 end
+
+using SymbolicUtils: walk
+
 function expand_derivatives(n::Num, simplify=false; kwargs...)
-    wrap(expand_derivatives(value(n), simplify; kwargs...))
+    n = walk(x -> x isa Float64 ? Rational(x) : x, value(n))
```

Comment:
```suggestion
    n = walk(x -> x isa Int? Rational(x) : x, value(n))
```

Is this sufficient to fix the issue?